### PR TITLE
[SYCL] Remove sycl_ext_oneapi_group_algorithms

### PR DESCRIPTION
**Pull request by aelovikov-intel**
Removed tests have SYCL 2020 versions in separate files under test-e2e/GroupAlgorithm.

### CODE REVIEW

**Comment by gmlueck** [comment]
I think this is goodness, but @Pennycook should probably give the approval from @intel/dpcpp-specification-reviewers.

**Review by Pennycook — APPROVED** [review]
Hurray! 🎉 

This looks good to me.

File: sycl/test-e2e/SubGroup/generic_reduce.cpp [hunk 1]
```diff
@@ -1,6 +1,12 @@
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 
+// SYCL 2020: Only the binary operators defined in Section 4.17.2 are supported
+// by the reduce functions in SYCL 2020, but the standard C++ syntax is used for
+// forward compatibility with future SYCL versions.
+//
+// REQUIRES: TEMPORARY_DISABLED
+
```
Comment (aelovikov-intel) [inline sycl/test-e2e/SubGroup/generic_reduce.cpp:9]: @Pennycook , are you ok with this?

Comment (Pennycook) [reply sycl/test-e2e/SubGroup/generic_reduce.cpp:9]: I'm not sure I understand why it's being temporarily disabled.

Does `reduce_over_group` (currently) have different type constraints than the old extension, or something like that?

Comment (aelovikov-intel) [reply sycl/test-e2e/SubGroup/generic_reduce.cpp:9]: From what I understand - yes.

Comment (Pennycook) [reply sycl/test-e2e/SubGroup/generic_reduce.cpp:9]: Then I'm okay with it.

**Review by cperkinsintel — APPROVED** [review]

**Pull request merged**